osu.Game/Overlays/MedalSplash/DrawableMedal.cs

```diff
@@ -51,7 +50,7 @@ public DrawableMedal(Medal medal)
                     Alpha = 0f,
                     Children = new Drawable[]
                     {
-                        medalSprite = new Sprite
+                        new DelayedLoadWrapper(() => new MedalOnlineSprite(medal), 0)
```

Comment:
Is this really what we want? Doesn't this mean the medal will display without a texture on slow internet? I would rather it were delayed (by LCA) as it was previously. The LCA and whole "preparing to display" stuff in `MedalOverlay` looks unnecessary now because it's not doing anything that would take a long time to load?

Reply:
> The LCA and whole "preparing to display" stuff in MedalOverlay looks unnecessary now because it's not doing anything that would take a long time to load?

It's still loading a sample and texture. If a stutter from that is not of concern, then I guess it could be removed, but I'd prefer having it there?

Reply:
I don't understand why this is better than having the whole overlay sit there loading? I would rather have the overlay _look good_ than to have it display immediately as soon as a medal is awarded, no?

Master:

https://github.com/user-attachments/assets/d17a1d13-1546-4dcb-9be4-246cfbba5a77

This PR:

https://github.com/user-attachments/assets/e6fc3d62-442b-41ed-b715-14e9681be9dd

> If a stutter from that is not of concern

Have you actually noticed a stutter there...? I'd be highly concerned if that caused a stutter because we do that pattern everywhere else in the game without LCA.

Reply:
Dunno. Perhaps I'm being dense with the above, so will ask for a second opinion from @bdach

Reply:
I mean, anywhere we are loading sprites or samples can cause a stutter, right? So in general we try and load these asynchronously?

I can revisit this, but it'll probably be a further refactor as the overlay simply cannot be present while the load is happening (right now on `master` it blocks the whole game for the *entire duration of the resource load*).

Reply:
> right now on `master` it blocks the whole game for the _entire duration of the resource load_

I might be talking out of rear here, but isn't that possibly fixable by delaying the show here?:

https://github.com/ppy/osu/blob/1e6c04e98b092e52a35b20d07e9a5a67e61de1b4/osu.Game/Overlays/MedalOverlay.cs#L155-L156

As in, move the `Show()` call to the LCA callback there.

My general opinion is that it'd obviously be better to not use DLW as it'll inevitably incur a higher risk ot things looking jank than ensuring everything is loaded prior to show, but I wouldn't necessarily insist on it if it turns out to be too annoying to make happen.

Reply:
Yeah, potentially that works. Just have to make sure the load actually still works without the thing loading being present. It would be the refactor I'd look at doing.

I'll give that direction a try and see how it goes. It kind of freaks me out that right now, a loading texture can block medals from ever displaying again (I'm not sure if we have an established "load failed" path for drawables?).

Reply:
I don't think that will work because the scheduler (LCA callback) is only updated if the overlay is present. But there's like 1001 different ways to get around that, not the least being to use the scheduler callback on LCA, set AlwaysPresent, or override IsPresent.

Reply:
To be honest if that's so hard to handle and DLW is used as a roundabout way to handle that, then I'm even less confident about this change. I'm not interested in quick merging this as it was reported once ever, so I would rather it be fixed properly as long as we agree that my dislike of DLW is not insane.

Also I'm not exactly sure what can "fail" here, `OnlineStore` try-catches everything and returns `null`, and so will `TextureStore`. I understand you haven't reproed it (and I haven't either - tested prior to this PR), but that feels like a weird thing to hinge on.